File: 06 Majority Element/Majority Element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function (nums) {
    let mp = new Map();
    let mx = -Infinity, value;
    for (let i = 0; i < nums.length; i++) {
        let x = 1;
        if (mp.has(nums[i]))
            x += mp.get(nums[i]);
        mp.set(nums[i], x);
        if (x > mx)
            mx = x, value = nums[i];
    }
    return value;
};

File: 06 Majority Element/Majority Element.cpp
class Solution
{
public:
    int majorityElement(vector<int> &nums)
    {
        map<int, int> mp;
        int mx = INT_MIN, value;
        for (int i = 0; i < nums.size(); i++)
        {
            mp[nums[i]]++;
            if (mp[nums[i]] > mx)
                mx = mp[nums[i]], value = nums[i];
        }
        return value;
    }
};